src/transformers/modeling_tf_bert.py

```diff
@@ -781,6 +781,8 @@ class TFBertForPreTrainingOutput(ModelOutput):
     BERT_START_DOCSTRING,
 )
 class TFBertModel(TFBertPreTrainedModel):
+    authorized_unexpected_keys = [r"nsp___cls"]
```

Comment:
Why do we need to add `"nsp__cls"` to `authorized_unexpected_keys` here? I'm not sure that we need it

Reply:
I think if one initializes a `TFBertModel` with `nsp` layer it is correct to show a warning that `nsp__cls` is not used because `nsp__cls` is not part of `TFBertMainLayer` in contrast to `pooler`.

Reply:
We don't need it, just in case it appears in the weights we don't raise the exception. Do you think it would be better to authorize none keys here?

Reply:
Fine, let's remove it then!